tests/test_plugin_loader.py

```
@@ -2411,6 +2654,117 @@ def test_stop_global_scheduler_no_thread(self, mock_threading):
+        )  # Function should return False on git operation failures
+        # Verify no "Invalid ref type" error was logged (commit ref type should be accepted)
+        for call in mock_logger.error.call_args_list:
+            self.assertNotIn("Invalid ref type", str(call))
+
+    @patch("mmrelay.plugin_loader._run_git")
+    @patch("mmrelay.plugin_loader._is_repo_url_allowed")
+    @patch("mmrelay.plugin_loader.logger")
+    @patch("os.path.isdir")
+    @patch("os.makedirs")
+    def test_clone_or_update_repo_new_repo_commit(
+        self, mock_makedirs, mock_isdir, mock_logger, mock_is_allowed, mock_run_git
```

Ruff (0.14.3)

Unused method argument: `mock_makedirs`

(ARG002)